tests/BanditClientTest.php

```diff
@@ -0,0 +1,272 @@
+        $subjectKey = 'user123';
+        $subject = ['country' => 'USA', 'age' => 25];
+        $default = 'defaultVariation';
+        $banditKey = $default;
+
+        $bandit = new Bandit(
+            $banditKey,
+            'falcon',
+            new DateTime(),
+            'v123',
+            new BanditModelData(
+                1.0,
+                [],
+                0.1,
+                0.1
+            )
+        );
+
+        $evaluation = new BanditEvaluation(
+            $flagKey,
+            $subjectKey,
+            AttributeSet::fromArray($subject),
+            'banditAction',
+            AttributeSet::fromArray([]),
+            200,
+            0.5,
+            1.0,
+            50
+        );
+        $expectedResult = new BanditResult('defaultVariation', 'banditAction');
+
+
+        $config = $this->getMockBuilder(ConfigurationLoader::class)->disableOriginalConstructor()->getMock();
+
+        $config->expects($this->once())
+            ->method('isBanditFlag')
+            ->with($flagKey)
+            ->willReturn(true);
+
+        // We know the assignment will evaluate to the default so let's use that shortcut to give us a bandit.
+        $config->expects($this->once())
+            ->method('getBanditByVariation')
+            ->with($flagKey, $default)
+            ->willReturn($banditKey);
+        $config->expects($this->once())
+            ->method('getBandit')
+            ->with($banditKey)
+            ->willReturn($bandit);
+
+        $banditEvaluator = $this->getMockBuilder(IBanditEvaluator::class)->getMock();
+        $banditEvaluator->expects($this->once())
+            ->method('evaluateBandit')
+            ->willReturn($evaluation);
+
+        $mockLogger = $this->getMockBuilder(LoggerInterface::class)->getMock();
+
+        // EppoClient won't log this assignment as it's not computed, just returning the default.
+        $mockLogger->expects($this->never())->method('logAssignment');
+
+        $mockLogger->expects($this->once())->method('logBanditAction')
+            ->with(
+                $this->callback(function (BanditActionEvent $bee) use ($flagKey, $subjectKey, $banditKey) {
+                    return $bee->banditKey == $banditKey &&
+                        $bee->subjectKey == $subjectKey &&
+                        $bee->flagKey == $flagKey &&
+                        $bee->action == 'banditAction';
+                })
+            );
+
+        $client = EppoClient::createTestClient(
+            $config,
+            poller: $this->getPollerMock(),
+            logger: $mockLogger,
+            banditEvaluator: $banditEvaluator
+        );
+
+        $result = $client->getBanditAction($flagKey, $subjectKey, $subject, $actions, $default);
+
+
+        $this->assertEquals($expectedResult, $result);
+    }
+
+    public function testRepoTestCases(): void
+    {
+        // Load all the test cases.
+        $testCases = $this->loadTestCases();
```

Comment:
consider also asserting this array is not empty, that way we don't get tests that "pass" without doing anything

(could also be done in `loadTestCases()`)

Reply:
done